Clean up the image building guide and rearrange environment docs a bit

choldgraf opened the pull request:
This PR cleans up a bit of the dynamic image building guide with the goal of making it easier to discover, and to include a bit more content. Along the way it also consolidates our environment docs under a single section so that it's less cluttered and easier to find them.

**Request for feedback**: It takes some inspiration from the DevSeed docs that @chuckwondo put together (🙏) - maybe he or @maxrjones would be willing to give this a quick look and let me know if something like this would have been useful for the VEDA team to either point to, or to just take an re-mix for their own needs? I'm trying to make documentation like this ease the burden on communities like VEDA who need to explain hub functionality (or to help them discover that it exists in the first place!)

- closes https://github.com/2i2c-org/docs/issues/251

chuckwondo commented on user/environment/dynamic-imagebuilding.md:
```diff
@@ -0,0 +1,97 @@
+### Step 1: Prepare a GitHub Repository
+
+1. Create a **public GitHub repository**.
+2. Add the following optional files (choose what you need):
```
```suggestion
2. Add one or more of the following optional files, based upon your needs:
```

chuckwondo commented on user/environment/dynamic-imagebuilding.md:
```diff
@@ -0,0 +1,97 @@
+   * `postBuild` (executable) – script to run after installation.
+   * `start` (executable) – script to run each time the container starts.
+
+   If you include a `Dockerfile`, then `repo2docker` uses this definition and ignores the above files and a regular Docker build is performed. See [](#image-building:learn-more) for links to resources to learn more.
```
```suggestion
   `repo2docker` dynamically generates a `Dockerfile` for you, based upon which of these files you include in your repo, so you don't have to deal with the intricacies of writing your own `Dockerfile`. For further information about these files, see the [repo2docker configuration file guide](https://repo2docker.readthedocs.io/en/latest/configuration/).

   Alternatively, if your repo includes a `Dockerfile`, then `repo2docker` uses it directly (ignoring any of the files above) and performs a regular Docker build.
   
   See [](#image-building:learn-more) for links to additional resources.
```

chuckwondo reviewed (commented):
Thanks @choldgraf. This looks great! I've made only 2 small suggestions for perhaps a bit more clarity around some of the wording regarding the recognized configuration files.

choldgraf commented:
OK I think that I've implemented all of @chuckwondo 's suggestions and have a sister PR in repo2docker. I nerd-sniped myself along the way and learned a bit about how to make the freshdesk widget pre-populate fields. So I've added a little docs guide to that too :-)

I think this is good to go unless there are objections! It depends on the r2d PR but I think it's OK if we have a broken link for a little bit until it gets merged.

maxrjones commented:
This is so great! 🎉 ❤️
